programs/cleos/httpc.cpp

```diff
@@ -89,17 +89,22 @@ namespace eosio { namespace client { namespace http {
          if(std::regex_search(header, match, clregex))
             response_content_length = std::stoi(match[1]);
       }
-      EOS_ASSERT(response_content_length >= 0, invalid_http_response, "Invalid content-length response");
 
       std::stringstream re;
-      // Write whatever content we already have to output.
-      response_content_length -= response.size();
-      if (response.size() > 0)
-         re << &response;
+      if( response_content_length >= 0 ) {
+         // Write whatever content we already have to output.
```

Comment:
Why is this part not required for the non-response_content_length branch?

Reply:
The logic used when the _Content-Length_ header was found, was to read the exact amount of bytes indicated in it or to fail (boost::asio::transfer_exactly), so the code calculated the remaining amount of bytes to read based on what was already read.

On the other branch, since the _Content-Length_ header isn't present and we will read all the response until the end there is no need to do that.

Reply:
We can put the partial output write outside the condition and always do a two step process to write the output

```c++
auto response_size = response.size();
if (response_size > 0)
  re << &response;

if( response_content_length >= 0 ) {
   response_content_length -= response_size;
 ....
} else { 
 ...
}

re << &response;
return re.str();

```

Reply:
From what I can tell, the
```
         if (response.size() > 0)
            re << &response;
```
can just simply be removed. Any left over content in `response` will be there when we `re << &response;` in either case